ToDoApp/ToDoApp/CreateTaskViewModel.swift

```diff
@@ -13,36 +13,65 @@ enum InputErrors: Identifiable, Hashable {
         hashValue
     }
 }
+enum CreateTaskViewMode {
+    case createNewTask
+    case editExistingTask(TaskModel)
+}
 @MainActor
 class CreateTaskViewModel: ObservableObject {
     // MARK: Internal Properties
     @Published var error: InputErrors?
+    @Published var title: String
+    @Published var deadline: Date
+    @Published private(set) var mode: CreateTaskViewMode?
```

Comment:
this property should be a constant since no one should modify it and it should not be optional since it is always provided and there's no case when we don't have this value

Reply:
This has been fixed.